src/jit/codegencommon.cpp

```diff
@@ -6385,6 +6388,8 @@ void            CodeGen::genPopCalleeSavedRegistersAndFreeLclFrame(bool jmpEpilo
             assert((calleeSaveSPDeltaUnaligned % 8) == 0); // It better at least be 8 byte aligned.
             calleeSaveSPDelta = AlignUp((UINT)calleeSaveSPDeltaUnaligned, STACK_ALIGN);
 
+            // Unlike other frameType, SP is already close to callee-save reg area
+            // since the local frame will be deleted below first. So just adjust alignment.
             calleeSaveSPOffset = calleeSaveSPDelta - calleeSaveSPDeltaUnaligned;
```

Comment:
To me, the comments (as they pre-existed as well as now) don't describe these cases well enough to make the code understandable. A few more words about the fact that the generated code is actually different for the last case (and how), would be useful.